tonic/src/transport/service/discover.rs

```diff
@@ -1,34 +1,36 @@
 use super::super::service;
 use super::connection::Connection;
 use crate::transport::Endpoint;
+use std::hash::Hash;
 use std::{
-    collections::VecDeque,
-    fmt,
     future::Future,
     pin::Pin,
     task::{Context, Poll},
 };
+use tokio::stream::Stream;
 use tower::discover::{Change, Discover};
 
-pub(crate) struct ServiceList {
-    list: VecDeque<Endpoint>,
-    connecting:
-        Option<Pin<Box<dyn Future<Output = Result<Connection, crate::Error>> + Send + 'static>>>,
-    i: usize,
+pub(crate) struct DynamicServiceStream<K: Hash + Eq + Clone + Unpin> {
+    changes: Box<dyn Stream<Item = Change<K, Endpoint>> + Unpin + Send + 'static>,
```

Comment:
if we make the change above as proposed then this can just be the concrete `mpsc::Reciever` type which should simplify this impl!

Reply:
done